public/apps/configuration/panels/user-list.test.tsx

```diff
@@ -0,0 +1,173 @@
+      const columns = getColumns('user1');
+      const usernameRenderer = columns[0].render as (usename: string) => JSX.Element;
+      const Container = (props: { username: string }) => usernameRenderer(props.username);
+      const result = shallow(<Container username={'user2'} />);
+
+      expect(result.find(EuiBadge).length).toBe(0);
+    });
+  });
+
+  describe('UserList', () => {
+    const mockCoreStart = {
+      http: 1,
+    };
+    const setState = jest.fn();
+    jest.spyOn(React, 'useState').mockImplementation((initValue) => [initValue, setState]);
+
+    it('render empty', () => {
+      const component = shallow(
+        <UserList
+          coreStart={mockCoreStart as any}
+          navigation={{} as any}
+          params={{} as any}
+          config={{} as any}
+        />
+      );
+
+      expect(component.find(EuiInMemoryTable).prop('items')).toEqual([]);
+    });
+
+    it('fetch data', () => {
+      jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());
+      shallow(
+        <UserList
+          coreStart={mockCoreStart as any}
+          navigation={{} as any}
+          params={{} as any}
+          config={{} as any}
+        />
+      );
+
+      expect(getUserList).toBeCalled();
+      expect(getAuthInfo).toBeCalled();
+    });
+
+    it('fetch data error', () => {
+      jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());
+      getUserList.mockImplementationOnce(() => {
+        throw new Error();
+      });
+      // Hide the error message
+      jest.spyOn(console, 'log').mockImplementationOnce(() => {});
+      shallow(
+        <UserList
+          coreStart={mockCoreStart as any}
+          navigation={{} as any}
+          params={{} as any}
+          config={{} as any}
+        />
+      );
+
+      // Expect error flag set to true
+      expect(setState).toBeCalledWith(true);
+    });
+
+    it('delete user', (done) => {
+      shallow(
+        <UserList
+          coreStart={mockCoreStart as any}
+          navigation={{} as any}
+          params={{} as any}
+          config={{} as any}
+        />
+      );
+      const deleteFunc = useDeleteConfirmState.mock.calls[0][0];
+
+      deleteFunc();
+
+      process.nextTick(() => {
+        expect(requestDeleteUsers).toBeCalled();
+        done();
+      });
+    });
+
+    it('delete user error', (done) => {
+      requestDeleteUsers.mockImplementationOnce(() => {
+        throw new Error();
+      });
+      // Hide the error message
+      const loggingFunc = jest.fn();
+      jest.spyOn(console, 'log').mockImplementationOnce(loggingFunc);
+      shallow(
+        <UserList
+          coreStart={mockCoreStart as any}
+          navigation={{} as any}
+          params={{} as any}
+          config={{} as any}
+        />
+      );
+      const deleteFunc = useDeleteConfirmState.mock.calls[0][0];
+
+      deleteFunc();
+
+      process.nextTick(() => {
+        expect(loggingFunc).toBeCalled();
+        done();
```

Comment:
What is this for?

Reply:
Expect logging the error.

Reply:
If you mean process.nextTick that is for awaiting the async function.